File: src/main/GO/leetcode/lists/topinterview/LeetCode387.go
package topinterview

func firstUniqChar(s string) int {
	if len(s) == 0 {
		return 0
	}
	mp := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		mp[s[i]]++
	}
	for i := 0; i < len(s); i++ {
		if v, ok := mp[s[i]]; ok {
			if v == 1 {
				return i
			}
		}
	}
	return -1
}

File: src/main/java/leetcode/lists/topinterview/LeetCode387.java
package leetcode.lists.topinterview;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/28 02:20
 */
public class LeetCode387 {

    public int firstUniqChar(String s) {
        int[] arr = new int[26];
        int n = s.length();
        for (int i = 0; i < n; i++) {
            arr[s.charAt(i) - 'a']++;
        }
        for (int i = 0; i < n; i++) {
            if (arr[s.charAt(i) - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
}
